components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -157,14 +173,80 @@ public void setUp() throws Exception {
+    public void testBuildIDToken(OAuthTokenReqMessageContext messageContext,
+                                 OAuth2AccessTokenRespDTO tokenRespDTO) throws Exception {
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        Object o = SignedJWT.parse(idToken).getJWTClaimsSet();
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                TestConstants.CLIENT_ID);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
+        }
+
     @Test
-    public void testBuildIDToken() throws Exception {
+    public void testAuthoBuildIDToken() throws Exception {
+
+        authzTokenReqDTO = new OAuth2AuthorizeReqDTO();
+        authzMessageContext = new OAuthAuthzReqMessageContext(authzTokenReqDTO);
+        authzTokenRespDTO = new OAuth2AuthorizeRespDTO();
+        authzTokenReqDTO.setTenantDomain(SUPER_TENANT_DOMAIN_NAME);
+        authzTokenReqDTO.setConsumerKey(TestConstants.CLIENT_ID);
+        authzTokenReqDTO.setIdpSessionIdentifier(TestConstants.IDP_ENTITY_ID_ALIAS);
+        authzTokenReqDTO.setUser(user);
+        authzTokenRespDTO.setAccessToken(TestConstants.ACCESS_TOKEN);
 
         RealmService realmService = IdentityTenantUtil.getRealmService();
         PrivilegedCarbonContext.getThreadLocalCarbonContext()
                 .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
         IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
-        defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        String authoIDToken = defaultIDTokenBuilder.buildIDToken(authzMessageContext, authzTokenRespDTO);
+        Assert.assertEquals(SignedJWT.parse(authoIDToken).getJWTClaimsSet().getAudience().get(0),
+                TestConstants.CLIENT_ID);
+        Assert.assertEquals(SignedJWT.parse(authoIDToken).getJWTClaimsSet().getIssuer(),
+                "https://localhost:9443/oauth2/token");
     }
 
+    @Test
+    public void testClientIDNotFoundException() throws Exception {
+
+        RealmService realmService = IdentityTenantUtil.getRealmService();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
+        IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
+        tokenReqDTO.setClientId(null);
+        try {
+            defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
```

Comment:
we need to assert here. The expected is to throw exeception. If it didn't throw any exception, then we need to capture it. Because the Assert in the catch will be executed only if there is Exception thrown